web/src/message_edit.ts

```diff
@@ -847,6 +847,9 @@ export function end_message_row_edit($row: JQuery): void {
     // Clean up the upload handler
     upload.deactivate_upload(upload.edit_config(row_id));
 
+    // Clearing preview area
+    clear_preview_area($row);
```

Comment:
The `clear_preview_area` method looks like it is being called from multiple paths, even for those where the edit UI is itself not open — The "Close" and "Copy and close" buttons in the "View original message" UI. So let's add a simple conditional check, checking if the `preview_mode` class is present on the `$row` element before calling this method.

Reply:
I’ve added a simple conditional check to ensure that the clear_preview_area method is only called when the preview_mode class is present on the $row element.

`if ($row.hasClass("preview_mode")) {
        clear_preview_area($row);
 }`

Thanks for the guidance! Let me know if there are any other improvements I can make.